lib/prop.rb

```diff
@@ -9,5 +10,6 @@ class << self
     extend Forwardable
     def_delegators :"Prop::Limiter", :read, :write, :configure, :configurations, :disabled, :before_throttle
     def_delegators :"Prop::Limiter", :throttle, :throttle!, :throttled?, :count, :query, :reset
+    def_delegators :"LeakeyBucket::Bucket", :leaky, :leaky!, :reset_bucket
```

Comment:
I think we need to figure out a way to integrate the leaky bucket feature into the existing `Prop::Limiter` class.

It makes sense to have a `Prop::LeakyBucket` class that implements the core leaky bucket math, but `Prop::Limiter` should have a leaky bucket option and then use this class to do the heavy lifting